### PR TITLE
Draft of blog post on benefits of state machines

### PR DESCRIPTION
**Pull request by laurakalbag**
Summarising the feedback from last week’s weekly question.

### CODE REVIEW

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+<Tweet id="1469042301783654417" />
+
+> I think that the biggest benefit comes from being able to use the same state machine across different frameworks. With XState, even if you just use React on your projects, you can share your state machine to your [insert framework name here] friends.
+
+## Maintaining complex app logic
+
+<Tweet id="1468992722686017540" />
+
+Pat’s blog post goes into fascinating depth on his work with Crucible. And the section on XState goes from realizing that he needed a state machine to moving much of their application logic into statecharts.
```
Comment (farskid) [inline content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:33]: should this be `move` instead?

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+> We didn't start Crucible using XState but by the end of the project it was one of our most valuable tools…
+>
+> …
+>
+> I was still new to statecharts and sort of overwhelmed by the whole idea so I needed to find a small feature I could prototype XState with where it would make sense and add recognizable value to a flow/process that could use more structure to it…
+>
+> …
+>
+> I built out the flow based on services state and UI state, I defined all the valid transitions and where they'd go, I wired up the inputs/outputs to it, and then I turned it loose on matchmaking in an actual client and it worked beautifully.
```
Comment (farskid) [inline content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:43]: I know we don't want to edit other's texts but does `services` need apostrophe?

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+[Read all of Pat’s post on his blog](https://tivac.com/2020/10/19/crucible-rip/).
+
+## Maintainability
+
+Maintainability is a popular feature of state machines, and Joshua is a fan.
+
+<Tweet id="1468958231028326419" />
+
+  Matt’s article on ‘[useState vs useReducer vs XState - Part 1: Modals](https://stately.ai/blog/usestate-vs-usereducer-vs-xstate-part-1-modals)’ shares examples of these different approaches and their maintainability. And he followed it up with his [React Finland talk, ‘Make legacy code delightful with statecharts’](https://www.youtube.com/watch?v=zll9uDQOOq0).
```
Comment (farskid) [inline content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:55]: Matt's article is an internal link and we're rendering as an external with`noreferrer nofollow`
image

Comment (farskid) [reply content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:55]: Reminds me that we need a NextLink component in MDX for internal links

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Getting to the core of your logic
+
+The discussion on our Discord focused on mindset and how state machines can help us better understand our app logic. Jan got the conversation started.
```
Comment (farskid) [inline content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:59]: Could we link to our Discord server here please?

**Review by farskid — COMMENTED** [review]
Left some suggestions

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx (hunk 1 above)
Comment (laurakalbag) [reply content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:33]: No, but it does read a bit confusingly. Will reword.

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx (hunk 2 above)
Comment (laurakalbag) [reply content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:43]: Or a “the”. I think some light editing for clarity might be ok.

File: content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx (hunk 4 above)
Comment (laurakalbag) [reply content/posts/2021-12-14-what-are-the-biggest-benefits-youve-had-from-using-state-machines.mdx:59]: I’ll link to the specific discussion, as we’ve got another link to the server at the end of the page, and we don’t want to get repetitive.

**Review by davidkpiano — APPROVED** [review]

**Pull request merged**